pkg/recipes/terraform/config/config_test.go

```diff
@@ -533,9 +534,54 @@ func Test_Save_InvalidWorkingDir(t *testing.T) {
 	testDir := filepath.Join("invalid", uuid.New().String())
 	envRecipe, resourceRecipe := getTestInputs()
 
-	tfconfig := New(testRecipeName, &envRecipe, &resourceRecipe)
+	tfconfig := New(context.Background(), testRecipeName, &envRecipe, &resourceRecipe, nil, nil)
 
 	err := tfconfig.Save(testcontext.New(t), testDir)
 	require.Error(t, err)
 	require.Equal(t, fmt.Sprintf("error creating file: open %s/main.tf.json: no such file or directory", testDir), err.Error())
 }
+
+func Test_getSecretStoreID(t *testing.T) {
```

Comment:
We should add unit tests for all the new functions.

Reply:
I dont if we can add unit for `getGitSource` func but i added for the other ones